components/script/dom/window.rs

```diff
@@ -989,6 +987,51 @@ impl Window {
         global_scope.constellation_chan().send(message).unwrap();
     }
 
+    /// https://html.spec.whatwg.org/multipage/#scroll-to-the-fragment-identifier
+    pub fn check_and_scroll_fragment(&self, fragment: &str) {
+        let doc = self.Document();
+        let body = doc.GetBody();
+        let target = doc.find_fragment_node(fragment);
+
+        // Step 1
+        doc.set_target_element(target.r().map(|e| &*e));
```

Comment:
It is here because it is a Option<Root<T>>. You can see a lot of usage of `.r()` for this.